implementations/rust/ockam/ockam_command/src/node/create.rs

```diff
@@ -351,6 +351,20 @@ async fn run_foreground_node(
         stop_node_on_eof(&mut ctx, &opts, &node_name).await?;
     }
 
+    let _ = tokio::spawn(stop_foreground_node(ctx, opts, node_name)).await?;
```

Comment:
Whether you use tokio or ctrlc to handle the signals, they can be handled in a separate thread and run in the background... or handled in the main thread blocking its execution. What approach to use will depend on the use case.

With tokio, if you `await` the spawned task, you will stop the execution there until the inner future completes. So, you just want to spawn the thread and continue: `tokio::spawn(stop_foreground_node(ctx, opts, node_name));`. In this case, you can get away without communicating back to the main thread because you can stop the context from wherever you want as long as you are in an async runtime.

With `ctrlc`, the handler is a simple closure (non-async), so you have to communicate back to the main thread where you will be able to stop the context. A side effect of this is that you should run the `stop_node_on_eof` in a background tokio task and set up the ctrlc signal in the main thread, similar to how it's shown in the [docs example](https://docs.rs/ctrlc/latest/ctrlc/#example). The reason of this is explained on the other comment.

And yeah, this issue doesn't require too many lines, but it's full of hidden details and traps 😆